arcgis10_mapping_tools/MapAction/MapAction/PageLayoutProperties.cs

```diff
@@ -18,6 +19,109 @@
+        /// <param name="isMapFrameMissing">Specifies whether the named Map Frame was found in the MXD 
+        /// and if that is the cause of the Exception being raised</param>
+        /// <param name="missingMapElements">A List of MapElements which were not found in the MXD. Can
+        /// be null if all MapElements are present</param>
+        /// <param name="duplicateMapElements">A List of duplicate MapElements which were found in the 
+        /// MXD. Can be null if no duplicate MapElements are present</param>
+        internal MapActionMapTemplateException(
+            string message, 
+            Exception inner,
+            string mxdPath,
+            bool isMapFrameMissing, 
+            List<MapElementNames> missingMapElements, 
+            List<MapElementNames> duplicateMapElements)
+            : base(message, inner)
+        {
+            this.m_mxdPath = mxdPath;
+            this.m_isMapFrameMissing = isMapFrameMissing;
+            this.m_missingMapElements = missingMapElements;
+            this.m_duplicateMapElements = duplicateMapElements;
+        }
+
+        /// <summary>
+        /// This constructor is best used when needing to specify missingMapElements and duplicateMapElements. The named MapFrame is assumed to be present.
+        /// </summary>
+        /// <param name="message">The base Exception Class message.</param>
+        /// <param name="mxdPath">The path to the MXD file which is the subject of the exception</param>
+        /// <param name="isMapFrameMissing">Specifies whether the named Map Frame was found in the MXD 
+        /// and if that is the cause of the Exception being raised</param>
+        internal MapActionMapTemplateException(string message, string mxdPath, bool isMapFrameMissing)
+            : this(message, null, mxdPath, isMapFrameMissing, null, null)
+        {}
+
+        /// <summary>
+        /// This constructor is best used when needing to specify missingMapElements and duplicateMapElements. The named MapFrame is assumed to be present.
+        /// </summary>
+        /// <param name="message">The base Exception Class message.</param>
+        /// <param name="mxdPath">The path to the MXD file which is the subject of the exception</param>
+        /// <param name="missingMapElements">>A List of MapElements which were not found in the MXD. Can
+        /// be null if all MapElements are present</param>
+        /// <param name="duplicateMapElements">A List of duplicate MapElements which were found in the 
+        /// MXD. Can be null if no duplicate MapElements are present</param>
+        internal MapActionMapTemplateException(string message, string mxdPath, List<MapElementNames> missingMapElements, List<MapElementNames> duplicateMapElements)
+            : this(message, null, mxdPath, false, missingMapElements, duplicateMapElements)
+        {}
+    }
+
     public static class PageLayoutProperties

```

Comment:
Absolutely! This class is probably number one in my target list for refactoring. But I think there are enough other changes in this PR that I thought in this case refactoring should be a separate branch/PR/piece of work.